quinn-proto/src/connection/transmit_buf.rs

```diff
@@ -75,6 +75,37 @@ impl<'a> TransmitBuf<'a> {
         self.len() == 0
     }
 
+    /// Returns the GSO segment size
+    ///
+    /// This is also the maximum size datagrams are allowed to be.  The first and last
+    /// datagram in a batch are allowed to be smaller however.  After the first datagram the
+    /// segment size is clipped to the size of the first datagram.
+    pub(super) fn segment_size(&self) -> usize {
```

Comment:
I guess? I don't love the LoC overhead this relative to the benefit this provide over an implicit/documentation contract that callers shouldn't be mutating `TransmitBuf`'s fields directly. It is a private API, of course, which we can trivially change over time.

Reply:
That's essentially a comment on this entire commit.  I do feel fairly strongly that the encapsulation matters, in Rust this means you have to write accessors and that's fine for me.  I don't think a few more lines of code are any overhead, it is even kind of nice you get to write a clear doc comment for it.

The encapsulation is a large part of this PR: the intricate logic of how all these fields relate to each other the invariants that need to be upheld are now entirely encoded in the 15 lines of `new_datagram_inner`.  And forcing non-mut access to the fields makes it easy to audit this.